### PR TITLE
feat: jobDSL build Docker image s pipelines

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

* It replaces the build Docker images pipelines 

## Why is it important?

We move to configuration based pipelines generated with jobDSL, every Docker image has its own pipeline at https://apm-ci.elastic.co/job/apm-shared/job/docker-images/ and all the configuration is at https://github.com/elastic/apm-pipeline-library/blob/main/.ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy, it unifies the way we build, test, and troubleshoot Docker images.

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]

File: .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy [hunk 1]
```diff
@@ -85,6 +85,126 @@ def dockerImages = [
...
+  [
+    name: "apm-integration-testing-aux",
+    repo: 'https://github.com/elastic/apm-integration-testing.git',
+    build_script: "make -C docker all-tests",
+    push_script: "make -C docker all-push",
+    push: true
+  ],
+  [
+    name: "oracle-instant-client",
+    repo: 'https://github.com/elastic/apm-integration-testing.git',
```
Comment (v1v) [inline .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy:190]: I guess, this is only need for making this work, but the apm-it repo is not used at all?

File: .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy [hunk 2]
```diff
@@ -85,6 +85,126 @@ def dockerImages = [
...
+    folder: "apps/test-plans"
+  ],
+  [
+    name: "apm-integration-testing",
+    repo: 'https://github.com/elastic/apm-integration-testing.git',
+    tag: "daily",
+    push: true
+  ],
+  [
+    name: "apm-integration-testing-aux",
```
Comment (v1v) [inline .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy:182]: What's this?

Comment (v1v) [reply .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy:182]: https://github.com/elastic/apm-pipeline-library/pull/1534/files#diff-596057068aaa2ba87378d9db4a5f48e3dd8bb52c02accb89b04331587aa09e4cL423-L431


maybe a comment about the aim of this ?

File: .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy (hunk 1 above)
Comment (kuisathaverat) [reply .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy:190]: This is used in the APM Java agent

File: .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy (hunk 2 above)
Comment (kuisathaverat) [reply .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy:182]: It compiles all the APM ITs Docker images

File: .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy (hunk 1 above)
Comment (kuisathaverat) [reply .ci/jobDSL/jobs/apm-ci/apm-shared/docker-images/build_docker_images.groovy:190]: C&P, I will add something to ignore the repo if is empty

**Pull request merged**